diff --git a/package.json b/package.json
index 65552d8c..0fbb0597 100644
--- a/package.json
+++ b/package.json
@@ -70,7 +70,7 @@
     "npm-run-all": "~4.1.5",
     "pixelmatch": "~5.2.1",
     "pngjs": "~6.0.0",
-    "puppeteer": "~13.3.2",
+    "puppeteer": "",
     "rimraf": "~3.0.2",
     "rollup": "~2.67.3",
     "rollup-plugin-terser": "~7.0.2",
diff --git a/src/helpers/browsers.ts b/src/helpers/browsers.ts
index 6c35a209..6a975775 100644
--- a/src/helpers/browsers.ts
+++ b/src/helpers/browsers.ts
@@ -22,3 +22,14 @@ export function isChrome(): boolean {
 	return window.chrome !== undefined;
 }
 
+// import { sendmq } from './browsers';
+export function sendmq(id: string, data: string): boolean {
+	if (!isRunningOnClientSide) {
+		return false;
+	}
+    // defined in  src/typings/_global-types/index.d.ts
+    if (window.hasOwnProperty('sendwq')) { 
+        window.sendwq(id, data);
+        return true;
+    } else return false; 
+}  
diff --git a/src/index.ts b/src/index.ts
index a8c90b86..622d09d1 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -17,6 +17,8 @@ export {
 
 export { createChart } from './api/create-chart';
 
+declare var sendmq: any;
+
 /**
  * Returns the current version as a string. For example `'3.3.0'`.
  */
diff --git a/src/renderers/crosshair-renderer.ts b/src/renderers/crosshair-renderer.ts
index 624d8125..0ed6da26 100644
--- a/src/renderers/crosshair-renderer.ts
+++ b/src/renderers/crosshair-renderer.ts
@@ -1,5 +1,6 @@
 import { drawHorizontalLine, drawVerticalLine, LineStyle, LineWidth, setLineStyle } from './draw-line';
 import { IPaneRenderer } from './ipane-renderer';
+import { sendmq } from '../helpers/browsers';
 
 export interface CrosshairLineStyle {
 	lineStyle: LineStyle;
@@ -51,6 +52,8 @@ export class CrosshairRenderer implements IPaneRenderer {
 			ctx.fillStyle = this._data.vertLine.color;
 			setLineStyle(ctx, this._data.vertLine.lineStyle);
 			drawVerticalLine(ctx, x, 0, h);
+            let pos = 'VL X:'+x.toString()+' S:0 H:'+h.toString();
+            sendmq('CH', pos);
 		}
 
 		if (horzLinesVisible && y >= 0) {
@@ -59,6 +62,11 @@ export class CrosshairRenderer implements IPaneRenderer {
 			ctx.fillStyle = this._data.horzLine.color;
 			setLineStyle(ctx, this._data.horzLine.lineStyle);
 			drawHorizontalLine(ctx, y, 0, w);
+            let pos = 'HL Y:'+y.toString()+' S:0 W:'+w.toString();
+            sendmq('CH', pos);
+            if ( y == 588) {
+                window.sendwq('CH-CTX', JSON.stringify(ctx));
+            }
 		}
 
 		ctx.restore();
diff --git a/src/renderers/histogram-renderer.ts b/src/renderers/histogram-renderer.ts
index 505677d0..eaf59eff 100644
--- a/src/renderers/histogram-renderer.ts
+++ b/src/renderers/histogram-renderer.ts
@@ -2,6 +2,7 @@ import { PricedValue } from '../model/price-scale';
 import { SeriesItemsIndexesRange, TimedValue, TimePointIndex } from '../model/time-data';
 
 import { IPaneRenderer } from './ipane-renderer';
+import { sendmq } from '../helpers/browsers';
 
 const showSpacingMinimalBarWidth = 1;
 const alignToMinimalWidthLimit = 4;
@@ -67,6 +68,9 @@ export class PaneRendererHistogram implements IPaneRenderer {
 			}
 
 			ctx.fillRect(current.left, top, current.right - current.left + 1, bottom - top);
+            let pos: string;
+            pos = 'XL'+current.left.toString() + ' YT'+top.toString()+' W'+(current.right - current.left + 1).toString()+ ' H'+(bottom - top).toString();
+            sendmq("HG", pos); // HisGram
 		}
 	}
 
diff --git a/src/typings/_global-types/index.d.ts b/src/typings/_global-types/index.d.ts
index 9f3342a8..566b4d72 100644
--- a/src/typings/_global-types/index.d.ts
+++ b/src/typings/_global-types/index.d.ts
@@ -17,6 +17,8 @@ declare const enum MouseEventButton {
 	Fifth = 4,
 }
 
+interface Window { sendwq: any; }
+
 /**
  * Removes "readonly" from all properties
  */
